app/client/src/routes/home.tsx

```diff
@@ -1,16 +1,511 @@
+  useEffect(() => {
+    if (content.status === 'failure') return setInputsLoaded(true);
+    if (content.status !== 'success') return;
+    getUrlInputs(getAbortSignal(), content.data.domainValues)
+      .then((initialInputs) => {
+        inputDispatch({ type: 'initialize', payload: initialInputs });
+      })
+      .catch((err) => {
+        console.error(`Error loading initial inputs: ${err}`);
+      })
+      .finally(() => setInputsLoaded(true));
+  }, [content, getAbortSignal]);
 
+  // Track non-empty values relevant to the current profile
+  const [queryParams, setQueryParams] = useState<URLQueryState>({});
+
+  // Update URL when inputs change
+  useEffect(() => {
+    if (!inputsLoaded) return;
+
+    // Get selected parameters, including multiselectable fields
+    const newQueryParams: URLQueryState = {};
+    Object.entries(inputState).forEach(([field, value]) => {
+      if (value == null || (Array.isArray(value) && !value.length)) return;
+
+      // Extract 'value' field from Option types
+      const flattenedValue = getInputValue(value);
+      if (
+        controlFields.includes(field) ||
+        (dataProfile &&
+          (
+            profiles[dataProfile.value].fields as ReadonlyArray<string>
+          ).includes(field))
+      ) {
+        newQueryParams[field] = flattenedValue;
+      }
+    });
+
+    window.location.hash = buildQueryString(newQueryParams);
+
+    setQueryParams(newQueryParams);
+  }, [dataProfile, inputState, inputsLoaded]);
+
+  const [introVisible, setIntroVisible] = useState(
+    !!JSON.parse(getLocalStorageItem('showIntro') ?? 'true'),
+  );
+  const [dontShowAgain, setDontShowAgain] = useState<boolean | null>(null);
+
+  useEffect(() => {
+    if (dontShowAgain === null) return;
+    setLocalStorageItem('showIntro', JSON.stringify(!dontShowAgain));
+  }, [dontShowAgain]);
+
+  if (content.status === 'pending') return <Loading />;
+
+  if (content.status === 'failure')
+    return (
+      <Alert type="error">
+        Expert Query is currently unavailable, please try again later.
+      </Alert>
+    );
+
+  if (content.status === 'success')
+    return (
+      <div>
+        {introVisible && (
+          <Summary heading="How to Use This Application">
+            <p>
+              Select a data profile, then build a query by selecting options
+              from the input fields.
+            </p>
+            <div className="display-flex flex-justify flex-wrap">
+              <Checkbox
+                checked={dontShowAgain ?? false}
+                label="Don't show again on this computer"
+                onChange={(_ev) => setDontShowAgain(!dontShowAgain)}
+                styles={['margin-right-1 margin-y-auto']}
+              />
+              <Button
+                onClick={() => setIntroVisible(false)}
+                styles={['margin-top-2']}
+              >
+                Close Intro
+              </Button>
```

Comment:
```suggestion
              <Button
                onClick={() => setIntroVisible(false)}
                className="usa-button margin-top-2"
              >
                Close Intro
              </Button>
```

Part of comment above. I don't think this component is necessary at this time.